File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {

        HashMap<String,Integer> hash=new HashMap<>();
        HashSet<String> res=new HashSet<>();
         
        System.out.println(hash);
        int l=10;
        for(int i=0;i<(s.length())-9;i++){
            String slicer=s.substring(i,l);
            if (hash.containsKey(slicer)){
               hash.put(slicer,hash.getOrDefault(slicer,0)+1);
                res.add(slicer);
            }
            else{
                hash.put(slicer,1);
               
            }
            l+=1;
        }
   
        List<String> res1=new ArrayList<>(res);

       
        return res1;
        
    }
}

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
      
        hash={}
        res=set()
        l=10
      

        for i in range(0,len(s)-9):
            slices=s[i:l]
            if slices not in hash:
                hash[slices]=1
            else:
                hash[slices]+=1
                res.add(slices)
            l+=1
        return list(res)
        